mirrord/operator/src/crd.rs

```diff
@@ -144,3 +144,23 @@ pub struct CopyTargetSpec {
     /// Ignored if [`Target`] is not [`Target::Deployment`].
     pub scale_down: bool,
 }
+
+/// Custom resource for policies that limit what mirrord features users can use.
+#[derive(CustomResource, Clone, Debug, Deserialize, Serialize, JsonSchema)]
+#[kube(
+    group = "operator.metalbear.co",
+    version = "v1",
+    kind = "MirrordPolicy",
+    root = "MirrordPolicyCrd",
+    namespaced
+)]
+pub struct MirrordPolicySpec {
+    /// Specify the targets for which this policy applies, in the pod/my-pod deploy/my-deploy
+    /// notation. Targets can be matched using `*` and `?` where `?` matches exactly one
+    /// occurrence of any character and `*` matches arbitrary many (including zero) occurrences
+    /// of any character. If `None`, the policy will apply for all targets.
+    pub target: Option<String>,
+
+    /// Forbid clients from stealing traffic from the policy target.
+    pub block_steal: Option<bool>,
```

Comment:
I think this needs to be more future proof i.e

i.e have the field named `disallow`
then it accepts list of stuff it disallows, which the list is an enum, where one of the enum variants is steal, http steal so user can disable one or another or both

Reply:
Can do, although [this is already future proof](https://docs.rs/kube/latest/kube/derive.CustomResource.html#versioning):
> _if you add new properties behind options, and simply mark old fields as deprecated, then you can safely roll schema out changes without bumping the version_

So I think it's ok to decide by which is the better UI.
Changing to list unless you tell me otherwise.

About http steal - 1. why would an organization want to block filtering? 2. Separate options for header and path filters, or does one string block them both?

Reply:
> where one of the enum variants is steal, http steal 

Do we realy want different variants for all possible steal rules (block all stealing, block unfiltered stealing, block filtered stealing, require header filter, require path filter, etc.), or do we want the steal variant to be an object itself so that users can specify like:

for blocking any stealing:
```yaml
disallow:
    - steal: any
```

for blocking unfiltered stealing:
```yaml
disallow:
    - steal: unfiltered
```

for blocking filtered stealing:
```yaml
disallow:
    - steal: filtered
```

If we just have a simple variant (string in the schema) for every mode users can create weird policies:
```yaml
disallow:
    - steal
    # redundant:
    - http steal
```

Reply:
I think the last option will be easiest, but I do understand what you mean about what you suggested.
The advantage of the last option is that you can then provide options to the specific - for example disallow specific ports or filter setc.